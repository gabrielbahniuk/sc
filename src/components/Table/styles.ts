import styled from 'styled-components'

export const Container = styled.div`
  margin-top: 4rem;
  display: flex;
  flex-direction: column;
  width: 100%;

  table {
    width: 100%;
    border-spacing: 0 0.25rem;

    thead,
    tbody {
      text-align: center;
    }
    th {
      background: var(--secondary-color);
      color: #101112;
      font-weight: 400;
      padding: 1rem 2rem;
      line-height: 1.5rem;
      cursor: pointer;
    }

    td {
      padding: 1rem 2rem;
      border: 0;
      background: #fff;
      color: #000;
    }
  }
`
