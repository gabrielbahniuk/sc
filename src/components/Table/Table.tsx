import React from 'react'
import useTimeSeries from '@/hooks/useTimeSeries'
import { appConfig } from '@/config'
import { formatCurrency } from '@/utils'

import { Container } from './styles'

type ConeProps = {
  riskLevel: number
  mu: number
  sigma: number
}

type TableProps = {
  yearsInvestment: number
  cones: ConeProps[]
  investmentValue: number
  riskLevel: number
  isVisible: boolean
}

const Table: React.FC<TableProps> = ({ investmentValue, yearsInvestment, cones, riskLevel, isVisible }) => {
  const { mu, sigma } = cones.filter((cone) => cone.riskLevel === riskLevel)[0]
  const { FEE, MONTHLY_SUM } = appConfig

  const { tableData } = useTimeSeries({
    years: yearsInvestment,
    initialSum: investmentValue,
    fee: FEE,
    mu,
    sigma,
    monthlySum: MONTHLY_SUM
  })

  const { labels, dataBad, dataGood, dataMedian } = tableData

  return (
    <>
      {isVisible && (
        <Container>
          <h1>Table</h1>
          <table>
            <thead>
              <tr>
                <th>Month</th>
                <th>Good</th>
                <th>Median</th>
                <th>Bad</th>
              </tr>
            </thead>
            <tbody>
              {labels.map((label, index) => {
                return (
                  <tr key={index}>
                    <td>{label}</td>
                    <td>{formatCurrency(dataGood[index])}</td>
                    <td>{formatCurrency(dataMedian[index])}</td>
                    <td>{formatCurrency(dataBad[index])}</td>
                  </tr>
                )
              })}
            </tbody>
          </table>
        </Container>
      )}
    </>
  )
}

export { Table }
