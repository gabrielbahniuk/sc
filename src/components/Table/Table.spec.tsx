import React from 'react'
import Table from '.'
import { create } from 'react-test-renderer'
import { shallow } from 'enzyme'

describe('Test Table component', () => {
  const tableProps = {
    investmentValue: 10,
    riskLevel: 3,
    yearsInvestment: 1,
    cones: [{ riskLevel: 3, mu: 0.0216, sigma: 0.0215 }]
  }
  it('should render Table if isVisible prop is true', () => {
    const table = create(<Table {...tableProps} isVisible={true} />).toJSON()
    expect(table).toMatchSnapshot()
  })
  it('should not render Table if isVisible prop is false', () => {
    const wrapper = shallow(<Table {...tableProps} isVisible={false} />)
    expect(wrapper.find('table')).toHaveLength(0)
  })
})
