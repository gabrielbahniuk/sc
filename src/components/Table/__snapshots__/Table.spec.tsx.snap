// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Table component should render Table if isVisible prop is true 1`] = `
<div
  className="sc-bdnxRM fYImHF"
>
  <h1>
    Table
  </h1>
  <table>
    <thead>
      <tr>
        <th>
          Month
        </th>
        <th>
          Good
        </th>
        <th>
          Median
        </th>
        <th>
          Bad
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          0
        </td>
        <td>
          €212.16
        </td>
        <td>
          €210.01
        </td>
        <td>
          €207.88
        </td>
      </tr>
      <tr>
        <td>
          1
        </td>
        <td>
          €416.18
        </td>
        <td>
          €410.21
        </td>
        <td>
          €404.33
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          €621.50
        </td>
        <td>
          €610.61
        </td>
        <td>
          €599.91
        </td>
      </tr>
      <tr>
        <td>
          3
        </td>
        <td>
          €827.93
        </td>
        <td>
          €811.20
        </td>
        <td>
          €794.80
        </td>
      </tr>
      <tr>
        <td>
          4
        </td>
        <td>
          €1,035.35
        </td>
        <td>
          €1,011.98
        </td>
        <td>
          €989.14
        </td>
      </tr>
      <tr>
        <td>
          5
        </td>
        <td>
          €1,243.68
        </td>
        <td>
          €1,212.96
        </td>
        <td>
          €1,183.00
        </td>
      </tr>
      <tr>
        <td>
          6
        </td>
        <td>
          €1,452.86
        </td>
        <td>
          €1,414.14
        </td>
        <td>
          €1,376.45
        </td>
      </tr>
      <tr>
        <td>
          7
        </td>
        <td>
          €1,662.84
        </td>
        <td>
          €1,615.50
        </td>
        <td>
          €1,569.52
        </td>
      </tr>
      <tr>
        <td>
          8
        </td>
        <td>
          €1,873.58
        </td>
        <td>
          €1,817.07
        </td>
        <td>
          €1,762.25
        </td>
      </tr>
      <tr>
        <td>
          9
        </td>
        <td>
          €2,085.07
        </td>
        <td>
          €2,018.82
        </td>
        <td>
          €1,954.69
        </td>
      </tr>
      <tr>
        <td>
          10
        </td>
        <td>
          €2,297.26
        </td>
        <td>
          €2,220.78
        </td>
        <td>
          €2,146.84
        </td>
      </tr>
      <tr>
        <td>
          11
        </td>
        <td>
          €2,510.15
        </td>
        <td>
          €2,422.92
        </td>
        <td>
          €2,338.73
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
